protocol/testutil/app/app.go

```diff
@@ -407,8 +412,12 @@ func (tApp *TestApp) AdvanceToBlock(
 	for tApp.App.LastBlockHeight() < int64(block) {
 		tApp.panicIfChainIsHalted()
 		tApp.header.Height = tApp.App.LastBlockHeight() + 1
-		// By default, only update block time at the requested block.
-		if tApp.header.Height == int64(block) {
+		if options.GradualBlockTimeIncrement {
```

Comment:
We want the last block to end up with the exact time the user choose so I would suggest making the `tApp.header.Height == int64(block)` check first so there is no rounding issue for the last block.